compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -129,7 +132,18 @@ abstract class AbstractKotlinSymbolProcessingExtension(val options: KspOptions,
             initialized = true
         }
         processors.forEach {
-            deferredSymbols[it] = it.process(resolver)
+            try {
+                deferredSymbols[it] = it.process(resolver)
+            } catch (e: Exception) {
+                // Throws KSP exceptions
+                if (e.stackTrace.first().className.startsWith("com.google.devtools.ksp")) {
+                    throw e
+                } else {
+                    val sw = StringWriter()
+                    e.printStackTrace(PrintWriter(sw))
+                    logger.error(sw.toString())
+                }
+            }
```

Comment:
We probably need to catch all calls to processors, such as `init`, `finish` and even `onError`.